tests/aggregation/tests.py

```diff
@@ -1434,6 +1434,17 @@ def test_aggregation_subquery_annotation_exists(self):
         )
         self.assertTrue(publisher_qs.exists())
 
+    def test_aggregation_filter_exists(self):
+        publishers_having_more_than_one_book_qs = (
+            Book.objects.values('publisher')
+                .annotate(cnt=Count('isbn'))
+                .filter(cnt__gt=1)
+        )
+        with self.assertNumQueries(1) as ctx:
+            self.assertTrue(publishers_having_more_than_one_book_qs.exists())
+        sql = ctx.captured_queries[0]["sql"]
+        self.assertNotIn(", (1)", sql)
```

Comment:
If we wanted to test it at the compiler level instead we could do something along the following

```suggestion
    def test_aggregation_filter_exists(self):
        queryset = (
            Book.objects.values('publisher')
                .annotate(cnt=Count('isbn'))
                .filter(cnt__gt=1)
        )
        query = queryset.query.exists(connection.alias)
        self.assertEqual(
            len(query.get_compiler(connection).pre_sql_setup()[2]), 1
        )
```

It heavily depends on implementation details of `django.db.models.sql` but avoids a query and reliance on backend specific output.

Reply:
Yeah it's better than `self.assertNotIn(", (1)", sql)` for sure

Reply:
Changed